src/sql/workbench/performance/browser/ADSPerformance.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface ADSPerformanceEntry {
+	readonly name: string;
+	readonly timestamp: number;
+}
+
+export class ADSPerformance {
+	static browserPerformance: Performance = window.performance;
+	static marks: ADSPerformanceEntry[] = [];
+	static transactionMarks: { string?: ADSPerformanceEntry[] } = {};
+	/**
+	 * Mark that something is taking place
+	 * @param name the event taking place
+	 */
+	public static mark(name: string, timestamp: number = ADSPerformance.getTimestamp()): void {
+		const entry: ADSPerformanceEntry = {
+			name,
+			timestamp
+		};
+
+		this.marks.push(entry);
```

Comment:
Instead of keeping our own list why not just use the ones stored in browserPerformance?

Reply:
In case we want to store some metadata in the future. For example the stack of where the code was called.

Reply:
Fair - but I'm slightly concerned since we'll have different timestamps than what browserPerformance is storing.

Reply:
> Fair - but I'm slightly concerned since we'll have different timestamps than what browserPerformance is storing.

This is true, but if we see it becomes an issue we can manipulate the timestamps for their entries. I don't really think this will be an issue as we __shouldn't__ be using their performance marks to create reports.